src/common/cluster-store.ts

```diff
@@ -64,6 +67,14 @@ export interface ClusterPreferences {
   httpsProxy?: string;
 }
 
+export interface ClusterRenderInfo extends ClusterModel {
```

Comment:
This feels a bit weird. Now we have `ClusterModel`, `ClusterState`, `Cluster`  and `ClusterRenderInfo` ... how is this different from `ClusterState`?

Reply:
`DeadError` and `name` mostly. I guess it is sort of weird but I don't see it as a big problem.

Reply:
I was also referring to `getByWorkspaceId` which now returns `ClusterRenderInfo[]` instead of `Cluster[]`. Would it make sense to include these fields in `ClusterState` and return `Cluster[]` as it was previously?

Reply:
I don't see why it would make sense for `DeadError` to be part of `ClusterState` let alone `Cluster` (see my responses to Lauri above). I guess `name` could be, but it doesn't really feel like a part of its state.

I can understand the desire for `getByWorkspaceId` to return `Cluster[]`. I will see what I can do with method overloading. I figured it would be fine if it didn't because the only two uses of the function were for rendering stuff, which is where I got the requirements for `ClusterRenderInfo` from.

Reply:
I didn't manage to get the overloading to work (maybe it is still possible and I am not skilled at typescript enough). So I just made a new function and kept `getByWorkspaceId` returning `Cluster[]`.

Reply:
I'm not really comfortable with this.. `ClusterStore` is also exposed to extension-api and `ClusterRenderInfo` is also leaked there... could we move `getRendererInfoByWorkspace` (and `ClusterRenderInfo`) out of this store, it's highly related to a single UI component?

Reply:
And the tray. `getWorkspaceById` (except for swapping icon order above) is only used in those two locations too.

I'll see if I can do that, maybe "protected" and a child class?